screens/Network/ContactGraph/ContactGraph.tsx

```diff
@@ -1,57 +1,72 @@
-import React, { useState } from "react"
-
-import {
-  View,
-  TextInput,
-  TouchableWithoutFeedback,
-  Keyboard,
-} from "react-native"
-
+import React, { useEffect, useState } from "react"
+import { View, TouchableWithoutFeedback, Keyboard, Image } from "react-native"
 import { styles } from "./styles"
-
 import Graph, {
   addContactNode,
   deleteNode,
   getNodeById,
   getNodes,
   Node,
-  setInitialized,
 } from "../../../components/Graph/Graph"
-
 import ForceDirectedGraph from "../../../components/Graph/ForceDirectedGraph/ForceDirectedGraph"
-
 import NodeModal from "../../../components/Graph/NodeModal/NodeModal"
-
 import { Contact } from "../../../types/Contact"
 import { getUserData } from "../../../firebase/User"
+import InputField from "../../../components/InputField/InputField"
 
 interface ContactGraphProps {
   onContactPress: (uid: string) => void
   graph: Graph
   userId: string
   userContact: Contact
+  loaded: boolean
+  navChange?: boolean
+  changeTab?: () => void
 }
 
 const ContactGraph = ({
   onContactPress,
   graph,
   userId,
   userContact,
+  loaded,
+  navChange,
+  changeTab,
 }: ContactGraphProps) => {
   const [searchText, setSearchText] = useState("")
   const [modalVisible, setModalVisible] = useState(false)
   const [clickedNode, setClickedNode] = useState<Node>(
     getNodeById(graph, userId)
   )
-  const [counter, setCounter] = useState(1)
 
-  const updateCounter = () => {
-    setCounter((prevCounter) => prevCounter + 1)
-  }
+  const [forceReload, setForceReload] = useState(false) // State variable for force reloading the graph
 
   const [magicNeighbors, setMagicNeighbors] = useState<string[]>([])
-
   const [magicPressedID, setMagicPressedID] = useState<string>("")
+  const [display, setDisplay] = useState(false)
+
+  useEffect(() => {
+    if (navChange && changeTab !== undefined) {
+      magicNeighbors.forEach((neighbor) => {
+        if (!userContact.friends?.includes(neighbor)) {
+          deleteNode(graph, neighbor)
+        }
+      })
+      setMagicNeighbors([])
+      setMagicPressedID("")
+      changeTab()
+    }
+  }, [navChange])
```

Comment:
There's also a github actions warning on this line too. I think you just need to include other variables in the sensitivity list of your useEffect, if they are relevant.

Reply:
No I don't this is on purpose